stubs/psycopg2/psycopg2/extras.pyi

```diff
@@ -28,22 +28,42 @@ from psycopg2._range import (
 
 from .extensions import connection as _connection, cursor as _cursor, quote_ident as quote_ident
 
+_T_cur = TypeVar("_T_cur", bound=_cursor)
+
 class DictCursorBase(_cursor):
     row_factory: Any
     def __init__(self, *args, **kwargs) -> None: ...
     def fetchone(self) -> tuple[Any, ...] | None: ...
     def fetchmany(self, size: int | None = ...) -> list[tuple[Any, ...]]: ...
     def fetchall(self) -> list[tuple[Any, ...]]: ...
-    def __iter__(self): ...
+    def __next__(self) -> tuple[Any, ...]: ...
 
 class DictConnection(_connection):
-    def cursor(self, *args, **kwargs): ...
+    @overload
+    def cursor(self, name: str | bytes | None = ..., *, withhold: bool = ..., scrollable: bool | None = ...) -> DictCursor: ...
+    @overload
+    def cursor(
+        self,
+        name: str | bytes | None = ...,
+        *,
+        cursor_factory: Callable[..., _T_cur],
+        withhold: bool = ...,
+        scrollable: bool | None = ...,
+    ) -> _T_cur: ...
+    @overload
+    def cursor(
+        self, name: str | bytes | None, cursor_factory: Callable[..., _T_cur], withhold: bool = ..., scrollable: bool | None = ...
+    ) -> _T_cur: ...
```

Comment:
These overloads are consistent with the base class's definitions of the `cursor` method except that the default `cursor_factory` case has been updated to return `DictCursor`. This comment also applies to `RealDictConnection` and `NamedTupleConnection`.